vertx-web/src/main/java/io/vertx/ext/web/Route.java

```diff
@@ -39,6 +41,27 @@
 @VertxGen
 public interface Route {
 
+  /**
+   * add metadata to this route. Used for saved extra data.
+   *
+   * @param key the metadata of key
+   * @param value the metadata of value
+   * @return a reference to this, so the API can be used fluently
+   */
+  @Fluent
+  default Route metadata(String key, Object value) {
```

Comment:
@wang007 usually we prefer builder like APIs, I'd suggest that instead of having this method we would have:

```java
@Fluent
Route putMetadata(String key, Object value);
```

That will put the given pair into the existing metadata map, or create a new metadata map and put, if null.